graph.h

```diff
@@ -61,9 +61,28 @@ class productGraph{
 class adjListVect{
 private:
     vector<pair<string,vector<string>>>adjVect;
+    friend class CSR;
 public:
     void addEdge(string edge1, string edge2);
     void buildPG(graph g, automata q);
     void results(string edge1);
     void printArr(vector<string> arr);
 };
+
+class CSR{
+private:
+    int currIndex=0;
+    int count=0;
+    vector<int> CSRmatrix;

```

Comment:
since we know how many vertices and edges are there, you can simply use int arrays to represent these two